bessctl/static/pipeline.js

```diff
@@ -0,0 +1,177 @@
+// colorscheme:
+// #01295f cool black
+// #437f97 queen blue
+// #849324 olive drab
+// #ffb30f dark tangerine
+// #fd151b vivid red
+
+// (node name, gate type, gate ID) -> [(timestamp, pkts, bits, cnt), ...]
+var stats = {};
+
+var opt_field;
+var opt_mode;
+
+function gates_to_str(gates, gate_type) {
+    var ret = '';
+
+    for (i = 0; i < gates.length; i++) {
+        gate_num = gates[i][gate_type]
```

Comment:
You are declaring variables like `gate_num` w/o the `var` keyword.  This is fine, but just be aware since you're not using `var`, the variable is global.

Reply:
You do this several times, not sure if intentional or not?

Reply:
Oh, of course not. I didn't even know that it becomes global. I thought `var` in js is something optional just like `;` without any semantic difference :+1: